torch/ao/quantization/fx/_lower_to_native_backend.py

```diff
@@ -402,19 +467,20 @@ def _lower_static_weighted_ref_module(
         # If so, we replace the entire fused module with the corresponding quantized module
         if ref_class in STATIC_LOWER_FUSED_MODULE_MAP:
             inner_ref_class, q_class = STATIC_LOWER_FUSED_MODULE_MAP[ref_class]
-            if type(ref_module[0]) != inner_ref_class:
+            if type(ref_module[0]) != inner_ref_class:  # type: ignore[index]
                 continue
```

Comment:
nit: this is also a matching step, I think we can move this closer to the match_static_pattern call or support this inside that function